hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -144,24 +144,36 @@ def __init__(self, element_strategy, shape, dtype, fill, unique):
             max_f4 = (2. - 2 ** -23) * 2 ** 127
             self.check_cast = lambda x: \
                 (not np.isfinite(x)) or (-max_f4 <= x <= max_f4)
+        elif dtype.kind == u'c' and dtype.itemsize == 8:
+            max_f4 = (2. - 2 ** -23) * 2 ** 127
+            self.check_cast = lambda x: (not np.isfinite(x)) or (
+                -max_f4 <= x.real <= max_f4 and -max_f4 <= x.imag <= max_f4
+            )
+        elif dtype.kind == u'U':
+            length = dtype.itemsize // 4
+            self.check_cast = \
+                lambda x: len(x) <= length and u'\0' not in x[length:]
+        elif dtype.kind in (u'S', u'a'):
+            self.check_cast = lambda x: len(x) <= dtype.itemsize and \
+                b'\0' not in x[dtype.itemsize:]
         else:
             self.check_cast = lambda x: True
 
     def set_element(self, data, result, idx, strategy=None):
         strategy = strategy or self.element_strategy
         val = data.draw(strategy)
+        result[idx] = val
```

Comment:
Why did this move up here?

Reply:
So that we can show the `repr(result[idx])` as well as `repr(val)` in the deprecation message, which is particularly useful if e.g. strings have been truncated.